### PR TITLE
Implement more advanced address checks

### PR DESCRIPTION
**Pull request by webmaster128**
This change becomes necessary for addresses that are not 45 characters long, i.e. when address prefix is not 6 chars long.

### CODE REVIEW

File: components/forms/TransactionForm.js [hunk 1]
```diff
@@ -56,23 +56,25 @@ class TransactionForm extends React.Component {
...
-      const res = await axios.post("/api/transaction", { dataJSON });
-      const { transactionID } = res.data;
-      this.props.router.push(
-        `${this.props.address}/transaction/${transactionID}`
-      );
-    } else {
-      this.setState({ addressError: "Use a valid cosmos-hub address" });
+    const addressError = checkAddress(this.state.toAddress);
+    if (addressError) {
+      this.setState({ addressError: `Invalid address for network ${process.env.NEXT_PUBLIC_CHAIN_ID}: ${addressError}` });
```
Comment (webmaster128) [inline components/forms/TransactionForm.js:61]: Removed the hardcoded cosmos-hub from this text

File: lib/displayHelpers.js [hunk 2]
```diff
@@ -91,4 +91,31 @@ const examplePubkey = (index) => {
...
+    return error.toString();
+  }
+
+  if (prefix !== process.env.NEXT_PUBLIC_ADDRESS_PREFIX) {
+    return `Expected address prefix '${process.env.NEXT_PUBLIC_ADDRESS_PREFIX}' but got '${prefix}'`;
+  }
+
+  if (data.length !== 20) {
+    return "Invalid address length in bech32 data. Must be 20 bytes.";
+  }
```
Comment (webmaster128) [inline lib/displayHelpers.js:116]: This is the length check we had before, but generic for all prefix lengths.

The 45 characters check we had before only works as long as the prefix has the same length (like "cosmos" and "tgrade").

File: lib/displayHelpers.js [hunk 3]
```diff
@@ -91,4 +91,31 @@ const examplePubkey = (index) => {
...
+const checkAddress = (input) => {
+  if (!input) return "Empty";
+
+  let data;
+  let prefix;
+  try {
+     ({ data, prefix } = Bech32.decode(input));
+  } catch (error) {
+    return error.toString();
+  }
```
Comment (webmaster128) [inline lib/displayHelpers.js:108]: This checks all sort of bech32 internals such as format and checksum

File: components/forms/TransactionForm.js (hunk 1 above)
Comment (samepant) [reply components/forms/TransactionForm.js:61]: ah nice catch

File: lib/displayHelpers.js (hunk 2 above)
Comment (samepant) [reply lib/displayHelpers.js:116]: 🔥

**Review by samepant — APPROVED** [review]

**Comment by samepant** [comment]
Looks good, really appreciate you catching these subtleties

**Pull request merged**